nowcasting_dataset/data_sources/sun/raw_data_load_save.py

```diff
@@ -0,0 +1,181 @@
+import datetime
+import time
+import io
+import logging
+from concurrent import futures
+from typing import List, Union, Optional
+
+import fsspec
+import numcodecs
+import pandas as pd
+from tqdm import tqdm
+import xarray as xr
+import numpy as np
+from pathlib import Path
+
+from nowcasting_dataset import geospatial
+
+logger = logging.getLogger(__name__)
+
+
+def get_azimuth_and_elevation(
+    datestamps: List[datetime.datetime], x_centers: List[int], y_centers: List[int]
+) -> (pd.DataFrame, pd.DataFrame):
+    """
+
+    Get Azimuth and elevation positions of the sun
```

Comment:
Nice docstring!

One tiny addition might be to make it explicit that `len(datestamps) == len(x_centers) == len(y_centers)`

Reply:
infact datestamps can be different length that the x,y locations. This means we can put in 1 year of datestamps, for various site locations